core/templates/pages/skill-editor-page/editor-tab/skill-prerequisite-skills-editor/skill-prerequisite-skills-editor.component.ts

```diff
@@ -13,123 +13,153 @@
-          for (var idx in $scope.skill.getPrerequisiteSkillIds()) {
-            if ($scope.skill.getPrerequisiteSkillIds()[idx] === skillId) {
-              AlertsService.addInfoMessage(
-                'Given skill is already a prerequisite skill', 5000);
-              return;
-            }
-          }
-          SkillUpdateService.addPrerequisiteSkill($scope.skill, skillId);
-        }, function() {
-          // Note to developers:
-          // This callback is triggered when the Cancel button is clicked.
-          // No further action is needed.
-        });
-      };
-
-      $scope.togglePrerequisiteSkills = function() {
-        if (WindowDimensionsService.isWindowNarrow()) {
-          $scope.prerequisiteSkillsAreShown = (
-            !$scope.prerequisiteSkillsAreShown);
-        }
-      };
-      ctrl.$onInit = function() {
-        groupedSkillSummaries = SkillEditorStateService
-          .getGroupedSkillSummaries();
-        TopicsAndSkillsDashboardBackendApiService.fetchDashboardDataAsync()
-          .then(function(response) {
-            categorizedSkills = response.categorizedSkillsDict;
-            untriagedSkillSummaries = response.untriagedSkillSummaries;
-          });
-
-        $scope.skill = SkillEditorStateService.getSkill();
-        $scope.prerequisiteSkillsAreShown = (
-          !WindowDimensionsService.isWindowNarrow());
-        $scope.skillIdToSummaryMap = {};
-
-        for (var name in groupedSkillSummaries) {
-          var skillSummaries = groupedSkillSummaries[name];
-          for (var idx in skillSummaries) {
-            $scope.skillIdToSummaryMap[skillSummaries[idx].id] =
-              skillSummaries[idx].description;
-          }
+import { Skill } from 'domain/skill/SkillObjectFactory';
+import { SkillUpdateService } from 'domain/skill/skill-update.service';
+import { SkillEditorStateService } from 'pages/skill-editor-page/services/skill-editor-state.service';
+import { AlertsService } from 'services/alerts.service';
+import { TopicsAndSkillsDashboardBackendApiService, TopicsAndSkillDashboardData } from 'domain/topics_and_skills_dashboard/topics-and-skills-dashboard-backend-api.service';
+import { WindowDimensionsService } from 'services/contextual/window-dimensions.service';
+import { NgbModal } from '@ng-bootstrap/ng-bootstrap';
+import { Component, EventEmitter, OnInit, Output } from '@angular/core';
+import { downgradeComponent } from '@angular/upgrade/static';
+
+@Component({
+  selector: 'oppia-skill-prerequisite-skills-editor',
+  templateUrl: './skill-prerequisite-skills-editor.component.html'
+})
+export class SkillPrerequisiteSkillsEditorComponent
+implements OnInit {
+  categorizedSkills: CategorizedSkills;
+  untriagedSkillSummaries: SkillSummary[];
+  groupedSkillSummaries: GroupedSkillSummaries;
+  skillIdToSummaryMap: object;
+  prerequisiteSkillsAreShown: boolean;
+  skill: Skill;
+  allAvailableSkills: SkillSummary[] = [];
+  @Output() updateView = new EventEmitter();
```

Comment:
Perhaps something like `onSkillIdsChanged` so that it follows the general pattern we follow for other event emitters.

Reply:
Has this been addressed? If so, please comment "done" or similar. Thanks!

Reply:
Done

Reply:
We follow convention of not prefixing `on` with Output bindings.

Reply:
Ah, I see, thanks for clarifying. Are there any patterns for naming event emitters?